src/dbus/server/dbus_thread_object.cpp

```diff
@@ -1522,6 +1524,21 @@ void DBusThreadObject::ActiveDatasetChangeHandler(const otOperationalDatasetTlvs
     SignalPropertyChanged(OTBR_DBUS_THREAD_INTERFACE, OTBR_DBUS_PROPERTY_ACTIVE_DATASET_TLVS, value);
 }
 
+void DBusThreadObject::LeaveNetworkHandler(DBusRequest &aRequest)
+{
+    mNcp->GetThreadHelper()->DetachGracefully([aRequest, this](otError error) mutable {
+        SuccessOrExit(error);
+        SuccessOrExit(error = otInstanceErasePersistentInfo(mNcp->GetThreadHelper()->GetInstance()));
+
+    exit:
+        aRequest.ReplyOtResult(error);
```

Comment:
Is this a synchronized call? Will the client receive the reply as exit is called right after this call.

Reply:
Probably not. Added flush to ensure reply is sent before exit.